### PR TITLE
All primary key related changes

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
- #47 
- #48 
- #50 
- #51 
- #76 
- #77 

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Comment by lfpll** [comment]
Hey great commit, I also investigated this problem and I don't think that will work, if you don't add the 'day' they will overwrite for users doing the same features in different days in a full refresh.

File: README.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ This tap:
 
 - US Subscription Endpoint: [https://app.pendo.io/api/v1/aggregation](https://app.pendo.io/api/v1/aggregation)
 - EU Subscription Endpoint: [https://app.eu.pendo.io/api/v1/aggregation](https://app.eu.pendo.io/api/v1/aggregation)
-- Primary key fields: `visitor_id`, `account_id`, `server`, `remote_ip`
+- Primary key fields: `feature_id`, `visitor_id`, `account_id`, `server`, `remote_ip`, `user_agent`, `day` or `hour`
```
Comment (sgandhi1311) [inline README.md:108]: I see the mismatch here - https://github.com/singer-io/tap-pendo/blob/pendo-pk-changes/tap_pendo/streams.py#L800

**Review by sgandhi1311 — COMMENTED** [review]
I think the readme.md needs to be updated with correct primary key fields. I see the difference in key_properties (stream.py) compared to fields mentioned on README.md

File: README.md [hunk 2]
```diff
@@ -114,7 +114,7 @@ This tap:
 
 - US Subscription Endpoint: [https://app.pendo.io/api/v1/aggregation](https://app.pendo.io/api/v1/aggregation)
 - EU Subscription Endpoint: [https://app.eu.pendo.io/api/v1/aggregation](https://app.eu.pendo.io/api/v1/aggregation)
-- Primary key fields: `visitor_id`, `account_id`, `server`, `remote_ip`
+- Primary key fields: `visitor_id`, `account_id`, `server`, `remote_ip`, `user_agent`, `day` or `hour`
```
Comment (sgandhi1311) [inline README.md:117]: Similar to above comment (feature_events)

File: README.md [hunk 3]
```diff
@@ -123,7 +123,7 @@ This tap:
 
 - US Subscription Endpoint: [https://app.pendo.io/api/v1/aggregation](https://app.pendo.io/api/v1/aggregation)
 - EU Subscription Endpoint: [https://app.eu.pendo.io/api/v1/aggregation](https://app.eu.pendo.io/api/v1/aggregation)
-- Primary key fields: `visitor_id`, `account_id`, `server`, `remote_ip`
+- Primary key fields: `page_id`, `visitor_id`, `account_id`, `server`, `remote_ip`, `user_agent`, `_sdc_parameters_hash`, `day` or `hour`
```
Comment (sgandhi1311) [inline README.md:126]: Similar to above comment (feature_events)

File: README.md [hunk 4]
```diff
@@ -150,7 +150,7 @@ This tap:
 
 - US Subscription Endpoint: [https://app.pendo.io/api/v1/aggregation](https://app.pendo.io/api/v1/aggregation)
 - EU Subscription Endpoint: [https://app.eu.pendo.io/api/v1/aggregation](https://app.eu.pendo.io/api/v1/aggregation)
-- Primary key fields: `visitor_id`, `account_id`, `server`, `remote_ip`
+- Primary key fields: `track_type_id`, `visitor_id`, `account_id`, `server`, `remote_ip`, `user_agent`, `day` or `hour`
```
Comment (sgandhi1311) [inline README.md:153]:
```suggestion
- Primary key fields: `track_type_id`, `visitor_id`, `account_id`, `remote_ip`, `user_agent`
```

File: README.md (hunk 1 above)
Comment (sgandhi1311) [reply README.md:108]:
```suggestion
- Primary key fields: `feature_id`, `visitor_id`, `account_id`, `remote_ip`, `user_agent`, `day` or `hour`
```

File: README.md (hunk 2 above)
Comment (sgandhi1311) [reply README.md:117]:
```suggestion
- Primary key fields: `visitor_id`, `account_id`, `remote_ip`, `user_agent`, `day` or `hour`
```

File: README.md (hunk 3 above)
Comment (sgandhi1311) [reply README.md:126]:
```suggestion
- Primary key fields: `page_id`, `visitor_id`, `account_id`, `remote_ip`, `user_agent`, `_sdc_parameters_hash`, `day` or `hour`
```

**Pull request merged**